PokeStar/PokeStar/DataModels/Raid.cs

```diff
@@ -1,131 +1,120 @@
 ﻿using System;
+using System.Linq;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using Discord.WebSocket;
 using PokeStar.ConnectionInterface;
 
 namespace PokeStar.DataModels
 {
    class Raid
    {
-      private const int playerLimit = 20;
-      private const int maxInvites = 10;
       public string Location { get; set; }
       public string Time { get; set; }
       public short Tier { get; set; }
       public RaidBoss Boss { get; private set; }
-      public int PlayerCount { get; private set; }
-      public int HereCount { get; private set; }
-      public Dictionary<SocketGuildUser, int> Attending { get; private set; }
-      public Dictionary<SocketGuildUser, int> Here { get; private set; }
-      public Dictionary<SocketGuildUser, int> Invite { get; private set; }
+      public List<RaidGroup> Groups { get; private set; }
+      public List<string> RaidBossSelections { get; set; }
       public DateTime CreatedAt { get; private set; }
+      private Dictionary<SocketGuildUser, int> Invite { get; set; }
 
       public Raid(short tier, string time, string location, string boss = null)
       {
          Tier = tier;
          Time = time;
          Location = location;
-         PlayerCount = 0;
-         HereCount = 0;
          SetBoss(boss);
-         Attending = new Dictionary<SocketGuildUser, int>();
-         Here = new Dictionary<SocketGuildUser, int>();
+         Groups = new List<RaidGroup>
+         {
+            new RaidGroup()
+         };
          Invite = new Dictionary<SocketGuildUser, int>();
+         RaidBossSelections = new List<string>();
          CreatedAt = DateTime.Now;
       }
 
-      public void PlayerAdd(SocketGuildUser player, int partySize, bool isInvite = false)
+      public ImmutableDictionary<SocketGuildUser, int> GetReadonlyInvite()
+      {
+         return Invite.ToImmutableDictionary(k => k.Key, v => v.Value);
+      }
+
+      public bool PlayerAdd(SocketGuildUser player, int partySize, SocketGuildUser user = null)
```

Comment:
whats the difference between player and user

Reply:
updated name of user to invitedby